### ISSUE TITLE
Make IPython an optional dependency

### ISSUE THREAD

**Issue by simonpercivall** [rkern/line_profiler#112]
It would be great if `line_profiler` didn't always require `IPython` to install, instead making it either a soft dependency, or an `extras_require` (with graceful degradation, of course).

**Comment by caethan** [comment]
Yes, we would love to do this.  I've been trying out ways to make it a soft dependency, but it's tricky with the way the IPython API has changed.  If you've got any suggestions, let us know!

**Comment by simonpercivall** [comment]
In its most simple way, I'd do it like this: PR #114

'Course, it'd be better to split out into a separate file, to make it look a bit cleaner.
